File: sample-app/src/main/java/com/psoffritti/slidingpanel/sampleapp/examples/utils/adapters/RecyclerViewAdapter.java
package com.psoffritti.slidingpanel.sampleapp.examples.utils.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import android.widget.Toast;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.psoffritti.slidingpanel.sampleapp.R;
import com.psoffritti.slidingpanel.sampleapp.examples.utils.DummyListItems;

import java.util.List;

public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewAdapter.ViewHolder> {

    private final List<DummyListItems.DummyItem> items;

    public RecyclerViewAdapter(List<DummyListItems.DummyItem> items) {
        this.items = items;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.recycler_view_sample_item, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull final ViewHolder holder, int position) {
        final DummyListItems.DummyItem item = items.get(position);
        holder.item = item;
        holder.view.setText(item.content);

        holder.view.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(v.getContext(), "Clicked: " +item.content, Toast.LENGTH_SHORT).show();
            }
        });
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        final TextView view;
        DummyListItems.DummyItem item;

        ViewHolder(View view) {
            super(view);
            this.view = (TextView) view;
        }
    }
}

File: sample-app/src/main/java/com/akbolatss/workshop/slidingpanel/sampleapp/utils/adapters/RecyclerViewAdapter.kt
package com.akbolatss.workshop.slidingpanel.sampleapp.utils.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import com.akbolatss.workshop.slidingpanel.sampleapp.R
import com.akbolatss.workshop.slidingpanel.sampleapp.utils.DummyListItems

class RecyclerViewAdapter(private val items: List<DummyListItems.DummyItem>) : RecyclerView.Adapter<RecyclerViewAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(
            LayoutInflater.from(parent.context).inflate(R.layout.recycler_view_sample_item, parent, false)
        )
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.item = items[position]
        holder.view.text = items[position].content

        holder.view.setOnClickListener {
            Toast.makeText(it.context, "Clicked: " + items[position].content, Toast.LENGTH_SHORT).show()
        }
    }

    override fun getItemCount(): Int {
        return items.size
    }

    inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val view: TextView = view as TextView
        var item: DummyListItems.DummyItem? = null

    }
}
